poc/six/student_loader.py

```diff
@@ -0,0 +1,330 @@
+using Data Standard 5.2. Supports insert, update, and delete operations.
+"""
+
+import csv
+import json
+import os
+import sys
+from datetime import datetime
+from typing import Dict, List, Optional
+import requests
+from dotenv import load_dotenv
+import pandas as pd
+
+
+class EdFiApiClient:
+    """Client for interacting with Ed-Fi API"""
+    
+    def __init__(self, base_url: str, client_id: str, client_secret: str):
+        self.base_url = base_url.rstrip('/')
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.access_token = None
+        self.session = requests.Session()
+        
+    def authenticate(self) -> bool:
+        """Authenticate with Ed-Fi API and get access token"""
+        auth_url = f"{self.base_url}/oauth/token"
+        auth_data = {
+            'grant_type': 'client_credentials',
+            'client_id': self.client_id,
+            'client_secret': self.client_secret
+        }
+        
+        try:
+            response = self.session.post(auth_url, data=auth_data)
+            response.raise_for_status()
+            token_data = response.json()
+            self.access_token = token_data['access_token']
+            self.session.headers.update({
+                'Authorization': f'Bearer {self.access_token}',
+                'Content-Type': 'application/json'
+            })
+            return True
+        except requests.exceptions.RequestException as e:
+            print(f"Authentication failed: {e}")
+            return False
+    
+    def create_student(self, student_data: Dict) -> bool:
+        """Create or update a student record"""
+        url = f"{self.base_url}/data/v3/ed-fi/students"
+        try:
+            response = self.session.post(url, json=student_data)
+            if response.status_code in [200, 201]:
+                print(f"✅ Student {student_data['studentUniqueId']} created/updated successfully")
+                return True
+            else:
+                print(f"❌ Failed to create/update student {student_data['studentUniqueId']}: {response.status_code} - {response.text}")
+                return False
+        except requests.exceptions.RequestException as e:
+            print(f"❌ Error creating/updating student {student_data['studentUniqueId']}: {e}")
+            return False
+    
+    def delete_student(self, student_unique_id: str) -> bool:
+        """Delete a student record"""
+        # First, get the student to find the ID
+        url = f"{self.base_url}/data/v3/ed-fi/students"
+        params = {'studentUniqueId': student_unique_id}
+        
+        try:
+            response = self.session.get(url, params=params)
+            if response.status_code == 200:
+                students = response.json()
+                if students:
+                    student_id = students[0]['id']
+                    delete_url = f"{self.base_url}/data/v3/ed-fi/students/{student_id}"
```

Comment:
It did not use the Discovery API for the base data management API URL. Again, must prompt for that.

The MCP server has a prompt that discusses this, which needs to be tuned, but Coding Agent apparently needs to be told explicitly about MCP prompts before it will use them.